addons/website/static/src/snippets/s_share/000.js

```diff
@@ -59,11 +59,18 @@ const ShareWidget = publicWidget.Widget.extend({
 
         const url = encodeURIComponent(window.location.href);
         const title = encodeURIComponent(document.title);
-        const media = encodeURIComponent(document.querySelector('meta[property="og:image"]').content);
+        const mediaEl = document.querySelector('meta[property="og:image"]');
+        let media = "";
+        if (mediaEl && mediaEl.content) {
+            media = encodeURIComponent(mediaEl.content);
+        }
 
         aEl.href = currentHref
             .replace(this.URL_REGEX, (match, a, b, c) => {
-                return a + url + c;
+                if (b === "{url}") {
+                    return a + url + c;
+                }
+                return a + b + c;
```

Comment:
Seemed louche since I don't get what would be the point of `this.URL_REGEX` if this was about just replacing `{url}` in the given string. Tested a bit: this actually breaks the whole purpose of [what was done here](https://github.com/odoo/odoo/commit/ddb2c06be4196c86ff6a2fd74874487d75b31c94) (which is what @rdeodoo linked as info in the task's chatter).

For once, let's close this PR and restart a new one with a new assignee from the team. We'll cherry-pick the relevant part of the work. But that way we won't keep disturbing here.

Reply:
Seemed louche since I don't get what would be the point of this.URL_REGEX if this was about just replacing {url} in the given string. Tested a bit: this actually breaks the whole purpose of [what was done here](https://github.com/odoo/odoo/commit/ddb2c06be4196c86ff6a2fd74874487d75b31c94) (which is what @rdeodoo linked as info in the task's chatter).

For once, let's close this PR and restart a new one with a new assignee from the team. We'll cherry-pick the relevant part of the work. But that way we won't keep disturbing here.

To be clearer about the flow that was broken:

    Go on to a product page (with variant)
    Click on the share by mail button
    => A sharer opens with the current URL to share, nice
    Close the popup and choose another variant
    Share again
    => You are still sharing the URL of the first variant

In fact, I don't remember if there is a good reason why we don't hardcode the whole sharers URL in JS and just replace the relevant part with a simple search and replace each time. To investigate.